modules/apps/list-type/list-type-service/service.xml

```diff
@@ -55,13 +55,18 @@
 		<finder name="ListTypeDefinitionId" return-type="Collection">
 			<finder-column name="listTypeDefinitionId" />
 		</finder>
+		<finder name="ERC_C" return-type="ListTypeEntry">
+			<finder-column name="externalReferenceCode"></finder-column>
+			<finder-column name="companyId"></finder-column>
```

Comment:
Why are you filtering by company and not for the "listTypeDefinitionId"?

Reply:
I think we need to scope by companyId + listTypeDefinitionId. 
(The object field is scoped like this).